### PR TITLE
TMA-1071: Reenable VCR

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/695/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1522/ : SUCCESS in 59s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1783/ : FAILURE in 59s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/22/ : SUCCESS in 5m 47s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/703/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1530/ : SUCCESS in 1m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1791/ : FAILURE in 1m 07s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/30/ : SUCCESS in 5m 50s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by petr-klemsinsky** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/708/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1535/ : FAILURE in 1m 18s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1796/ : FAILURE in 1m 18s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/31/ : FAILURE in 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/711/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1538/ : FAILURE in 2m 19s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1799/ : FAILURE in 2m 09s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/33/ : SUCCESS in 6m 04s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/712/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1539/ : FAILURE in 1m 24s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1800/ : SUCCESS in 3m 06s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/34/ : SUCCESS in 6m 14s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/713/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1540/ : SUCCESS in 2m 57s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1801/ : FAILURE in 1m 38s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/35/ : SUCCESS in 6m 24s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/719/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1543/ : FAILURE in 1m 28s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1804/ : SUCCESS in 1m 28s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/36/ : SUCCESS in 6m 33s

File: lib/gooddata/helpers/global_helpers.rb [hunk 1]
```diff
@@ -231,6 +231,8 @@ def encrypt(data, key)
       end
 
       def decrypt(database64, key)
+        vcr_mode_none = ENV['VCR_ON'] == 'true' && (!ENV['VCR_RECORD_MODE'] || ENV['VCR_RECORD_MODE'] == 'none') # TODO: refactor
```
Comment (yenkins) [inline lib/gooddata/helpers/global_helpers.rb]: info: Calls 'env['vcr_record_mode']' 2 times (DuplicateMethodCall) (reek)

File: spec/vcr_configurer.rb [hunk 2]
```diff
@@ -21,6 +21,16 @@ def self.filter_body_field(vcr_config, part, field)
         end
       end
 
+      def self.filter_sso(vcr_config)
```
Comment (yenkins) [inline spec/vcr_configurer.rb]: info: Has approx 7 statements (TooManyStatements) (reek)

File: spec/vcr_configurer.rb [hunk 3]
```diff
@@ -21,6 +21,16 @@ def self.filter_body_field(vcr_config, part, field)
...
+        part = 'request'
+        placeholder = name_to_placeholder('PGP')
+        vcr_config.filter_sensitive_data(placeholder) do |interaction|
+          login = interaction[part].uri.end_with? '/gdc/account/customerlogin'
+          require 'pry';binding.pry if login
```
Comment (yenkins) [inline spec/vcr_configurer.rb]: warning: Layout/SpaceAfterSemicolon: Space missing after semicolon. (rubocop)

Comment (yenkins) [inline spec/vcr_configurer.rb]: warning: Style/Semicolon: Do not use semicolons to terminate expressions. (rubocop)

Comment (yenkins) [inline spec/vcr_configurer.rb]: warning: Lint/Debugger: Remove debugger entry point `binding.pry`. (rubocop)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/722/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1546/ : FAILURE in 1m 49s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1807/ : SUCCESS in 1m 49s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/39/ : SUCCESS in 6m 18s

File: lib/gooddata/helpers/global_helpers.rb (hunk 1 above)
Comment (yenkins) [inline lib/gooddata/helpers/global_helpers.rb]: error: Identical code found in global_helpers.rb:234, connection_spec.rb:21 (mass = 68) (flay)

File: spec/integration/connection_spec.rb [hunk 4]
```diff
@@ -17,6 +17,9 @@
     end
 
     `gpg --import #{KEYS.join(' ')}`
+
+    vcr_mode_none = ENV['VCR_ON'] == 'true' && (!ENV['VCR_RECORD_MODE'] || ENV['VCR_RECORD_MODE'] == 'none') # TODO: refactor
```
Comment (yenkins) [inline spec/integration/connection_spec.rb]: error: Identical code found in global_helpers.rb:234, connection_spec.rb:21 (mass = 68) (flay)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/724/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1548/ : FAILURE in 1m 44s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1809/ : FAILURE in 1m 46s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/41/ : SUCCESS in 6m 30s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/725/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1549/ : FAILURE in 1m 50s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1810/ : SUCCESS in 2m 03s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/42/ : SUCCESS in 6m 20s

File: spec/environment/environment.rb [hunk 5]
```diff
@@ -23,6 +23,9 @@ def load(env = ENV['GD_ENV'] || 'testing')
         GoodData::Environment::ConnectionHelper.set_const('SECRETS', env_secrets)
         GoodData::Environment::ProjectHelper.set_const :PROJECT_URL, "/gdc/projects/#{GoodData::Environment::ProjectHelper::PROJECT_ID}"
         ENV['GD_SERVER'] = GoodData::Environment::ConnectionHelper::DEFAULT_SERVER
+        # VCR is enabled by default - set VCR_ON=false to disable
+        vcr_on = ENV['VCR_ON'].nil? || ENV['VCR_ON'].downcase == 'true'
```
Comment (yenkins) [inline spec/environment/environment.rb]: info: Calls 'env['vcr_on']' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline spec/environment/environment.rb]: info: Performs a nil-check (NilCheck) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/727/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1550/ : SUCCESS in 1m 48s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1811/ : FAILURE in 1m 41s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/43/ : SUCCESS in 6m 08s

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -38,37 +38,34 @@
...
     require 'vcr_configurer'
-    skip_sleep = VcrConfigurer.vcr_record_mode == :none
+    skip_sleep = GoodData::Helpers::VcrConfigurer.vcr_record_mode == :none
+
+    base = Pathname(__FILE__).dirname.expand_path
+    Dir.glob(base + 'helpers/*_helper.rb').each do |file|
+      require file
+    end
+
+    include GoodData::Helpers
```
Comment (yenkins) [inline spec/spec_helper.rb]: warning: Style/MixinUsage: `include` is used at the top level. Use inside `class` or `module`. (rubocop)

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -83,14 +80,21 @@ def peach_with_index(*, &y)
...
+          # avoid polling idle time by overriding sleep
           module Kernel
             alias :old_sleep :sleep
             def sleep(n)
               n
             end
           end
+
+          # make vcr tests work without password
+          module GoodData::Helpers
```
Comment (yenkins) [inline spec/spec_helper.rb]: warning: Style/ClassAndModuleChildren: Use nested module/class definitions instead of compact style. (rubocop)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/731/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1554/ : FAILURE in 1m 21s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1815/ : FAILURE in 1m 20s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/47/ : SUCCESS in 6m 13s

File: lib/gooddata/helpers/global_helpers.rb [hunk 8]
```diff
@@ -231,7 +231,7 @@ def encrypt(data, key)
       end
 
       def decrypt(database64, key)
-        return '' if key.nil? || key.empty?
+        return 'no_key_provided' if key.nil? || key.empty?
```
Comment (yenkins) [inline lib/gooddata/helpers/global_helpers.rb]: info: Performs a nil-check (NilCheck) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/732/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1555/ : SUCCESS in 1m 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1816/ : FAILURE in 1m 26s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/48/ : SUCCESS in 6m 32s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/733/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1556/ : SUCCESS in 1m 42s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1817/ : FAILURE in 1m 31s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/49/ : SUCCESS in 6m 29s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/734/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1557/ : SUCCESS in 1m 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1818/ : FAILURE in 1m 23s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/50/ : SUCCESS in 6m 27s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/737/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1560/ : FAILURE in 1m 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1821/ : SUCCESS in 1m 34s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/53/ : SUCCESS in 6m 04s

**Comment by panjan** [comment]
Integratino tests: https://travis-ci.org/panjan/gooddata-ruby/builds/440094703

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/740/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1563/ : SUCCESS in 1m 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1824/ : FAILURE in 1m 23s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/56/ : SUCCESS in 6m 32s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Pull request merged**